enterprise_subsidy/apps/fulfillment/api.py

```diff
@@ -32,3 +37,103 @@ def determine_fulfillment_client(subsidy_uuid, content_key):
+        return float(transaction.quantity) / self.CENTS_PER_DOLLAR
+
+    def _get_enterprise_customer_uuid(self, transaction):
+        return transaction.ledger.subsidy.enterprise_customer_uuid
+
+    def _get_geag_variant_id(self, transaction):
+        ent_uuid = self._get_enterprise_customer_uuid(transaction)
+        return ContentMetadataApi().get_geag_variant_id(ent_uuid, transaction.content_key)
+
+    def _create_allocation_payload(self, transaction, currency='USD'):
+        transaction_price = self._get_geag_transaction_price(transaction)
+        return {
+            'payment_reference': transaction.uuid,
+            'enterprise_customer_uuid': self._get_enterprise_customer_uuid(transaction),
+            'currency': currency,
+            'order_items': [
+                {
+                    # productId will be the variant id from product details
+                    'productId': self._get_geag_variant_id(transaction),
+                    'quantity': 1,
+                    'normalPrice': transaction_price,
+                    'discount': 0.0,
+                    'finalPrice': transaction_price,
+                }
+            ],
+            'first_name': transaction.metadata.get('geag_first_name'),
+            'last_name': transaction.metadata.get('geag_last_name'),
+            'date_of_birth': transaction.metadata.get('geag_date_of_birth'),
+            'terms_accepted_at': transaction.metadata.get('geag_terms_accepted_at'),
+        }
+
+    def _validate(self, transaction):
+        """
+        Validates that a ledger transaction contains all the required information to
+        construct a GEAG allocation payload.
+
+        Raises an exception when the transaction is missing required information
+        """
+        for field in ['geag_first_name', 'geag_last_name', 'geag_date_of_birth', 'geag_terms_accepted_at']:
+            if not transaction.metadata.get(field):
+                raise InvalidFulfillmentMetadataException(f'missing {field} transaction metadata')
+        return True
+
+    def _save_fulfillment_reference(self, transaction, external_reference_id):
+        external_fulfillment_provider, _ = ExternalFulfillmentProvider.objects.get_or_create(
+            slug=self.EXTERNAL_FULFILLMENT_PROVIDER_SLUG
+        )
+        return ExternalTransactionReference.objects.create(
+            transaction=transaction,
+            external_fulfillment_provider=external_fulfillment_provider,
+            external_reference_id=external_reference_id,
+        )
+
+    def can_fulfill(self, transaction):
+        """
+        A helper to let callers know if the transaction at hand can be fulfilled
+        with this fulfillment handler.
+        """
+        return self._get_geag_variant_id(transaction)
+
+    def _fulfill_in_geag(self, allocation_payload):
+        geag_response = self.get_smarter_client.create_enterprise_allocation(**allocation_payload)
+        return geag_response.json().get('orderUuid')
```

Comment:
[nit/curious/suggestion] Is there any harm in returning the full response JSON here? That is, if/when the GEAG allocation response payload includes more than just the `orderUuid`, we wouldn't have to come back to update this return.